include/deal.II/base/vectorization.h

```diff
@@ -4110,6 +4110,27 @@ namespace std
 
 
 
+  /**
+   * Variant of above function that takes an integer as second argument.
+   *
+   * @relatesalso VectorizedArray
+   */
+  template <typename Number, int width>
+  inline ::dealii::VectorizedArray<Number, width>
+  pow(const ::dealii::VectorizedArray<Number, width> &x, const int p)
+  {
+    Number values[::dealii::VectorizedArray<Number, width>::n_array_elements];
+    for (unsigned int i = 0;
+         i < dealii::VectorizedArray<Number, width>::n_array_elements;
+         ++i)
+      values[i] = std::pow(x[i], p);
```

Comment:
Since this is an integer power, wouldn't it make sense with some specific integer code path? I know this probably duplicates some code that might be contained in the standard library, but it is reasonably simple nonetheless and, if inlined, should remove all loops and conditional branches.
```C++
  template <typename Number, int width>
  inline ::dealii::VectorizedArray<Number, width>
  pow(const ::dealii::VectorizedArray<Number, width> &x, const int p)
  {
    unsigned int positive_exponent = std::abs(p);
    VectorizedArray<Number> result = Number(1);
    VectorizedArray<Number> x_to_p = x;
    while (true)
      {
        // exponent is odd -> this happens at least once if exponent>0,
        // so the result is ok
        if (positive_exponent & 1)
          result *= x_to_p;
        positive_exponent /= 2;
        if (positive_exponent == 0)
         break;
        x_to_p *= x_to_p;
      }
    if (p > 0)
      return result;
    else
      return Number(1)/result;
  }
```
```

Reply:
I should add that this way we do fully vectorized computations, so they should potentially be much faster.

Reply:
Good point!

Reply:
Don't we already have some `int_power` function somewhere that does essentially this, just for scalars?

Reply:
Yes, we have `Utilities::pow` which currently only works for non-negative exponents.

Reply:
@tamiko how much more functionality do you need? @masterleinad I guess we could switch to a templated `Utilities::pow` function, where we currently only support `int` types. I would see it work also for negative integer exponents at the expense of yet another `operator?:`. At the same time, the function body here is not too long, and it would give us the integer `pow` in the `std` namespace`.

Reply:
I didn't mean to suggest to unify code paths, but just to copy the code that has already worked well in another location. (Maybe with a reference to where the code was copied from.)

Reply:
@kronbichler Only the posted functionality for the time being.